### PR TITLE
support http/2 client response headers for GithubPRPlugin check

### LINKED ISSUE #407: HTTP/2 201 fails PR Posting
If the response uses http/2 the jobs fail to move past even though it was successful in posting the comment back to the PR. 

https://github.com/manheim/terraform-pipeline/blob/a14c20db5eca4587d025de4e3ca7b093700819a5/src/GithubPRPlanPlugin.groovy#L182

Fix should be to check for `HTTP/2 201` as well

### PR DESCRIPTION
**Pull request by codezninja**
## Merge Checklist

* [X] Have you linked this Pull Request to an [Issue](https://github.com/manheim/terraform-pipeline/issues)?
* [ ] Have you updated any relevant README files, to explain how this new feature works (with examples)?
* [X] Have you added relevant test cases for your change?
* [X] Do all tests and code style checks pass with `./gradlew check --info`?
* [ ] Have you successfully run your change in a pipeline in a Jenkins instance?
* [x] Have you updated the [CHANGELOG](https://github.com/manheim/terraform-pipeline/blob/master/CHANGELOG.md)?

Fixes #407

### CODE REVIEW

File: src/GithubPRPlanPlugin.groovy [hunk 1]
```diff
@@ -179,7 +179,7 @@ class GithubPRPlanPlugin implements TerraformPlanCommandPlugin, TerraformEnviron
                 def output = sh(script: cmd, returnStdout: true).trim()
 
                 def headers = readFile('comment.headers').trim()
-                if (! headers.contains('HTTP/1.1 201 Created')) {
+                if (! (headers.contains('HTTP/1.1 201 Created') || headers.contains('HTTP/2 201 Created'))) {
```
Comment (kmanning) [inline src/GithubPRPlanPlugin.groovy:182]: Do we really care about the HTTP version?  Could we simplify this to "201 Created"?

Not a change that you need to make, but arguably in the future, we could even git rid of the comment.headers, with a 

`[ curl -o output -I -L -s -w "%{http_code}" "${pullRequestUrl}" -eq 201 ]`

**Comment by bondezbond** [comment]
need to merge #410 before we can fix this

File: src/GithubPRPlanPlugin.groovy (hunk 1 above)
Comment (bondezbond) [reply src/GithubPRPlanPlugin.groovy:182]: I agree we can definitely simplify and check that. @jleopold28 thoughts?

Comment (jleopold28) [reply src/GithubPRPlanPlugin.groovy:182]: @bondezbond checking response code sounds good to me 👍

Comment (bondezbond) [reply src/GithubPRPlanPlugin.groovy]: done

Comment (bondezbond) [reply src/GithubPRPlanPlugin.groovy]: @kmanning during testing a colleague saw that http/2 only returns `201` now and drops `Created` part in response. So for now we'll have to to keep that complex conditional or we can move to a simplified curl call during this PR.

**Review by kmanning — APPROVED** [review]
:shipit:

**Pull request merged**